src/Command/FixMediaContextCommand.php

```diff
@@ -65,7 +64,6 @@ public function execute(InputInterface $input, OutputInterface $output): int
         }
 
         foreach ($this->mediaPool->getContexts() as $context => $contextAttrs) {
-            /** @var ContextInterface $defaultContext */
```

Comment:
The code below, the `getRootCategory` throws the same exception for several cases. When there is no rootCategory it throws an exception, we could try catch it, or implement hasRootCategory

Reply:
And the CategoryInterface -> getContext() always returns a Context, but that is not true, it can return null.

Reply:
The `getRootCategory` is deprecated and will be removed in 4.0 (even if there is no NEXT_MAJOR comment).
Seems like it the TODO was introduced here: https://github.com/sonata-project/SonataClassificationBundle/pull/287.

I'll try to resolve this https://github.com/sonata-project/SonataClassificationBundle/pull/684

I think you should now use `getRootCategoriesForContext`.

Reply:
Fixed